File: copilot-demo-labs/use-cases/10-security-analysis/src/AuthService.py

import hashlib


class AuthService:
    def __init__(self) -> None:
        self._users = {
            "admin": self._md5("admin123"),
        }

    def login(self, username: str, password: str) -> bool:
        if username not in self._users:
            return False

        return self._users[username] == self._md5(password)

    def issue_token(self, username: str) -> str:
        return f"token-{username}-12345"

    @staticmethod
    def _md5(value: str) -> str:
        return hashlib.md5(value.encode("utf-8")).hexdigest().upper()

File: copilot-demo-labs/use-cases/10-security-analysis/src/AuthService.cs
using System;
using System.Collections.Generic;
using System.Security.Cryptography;
using System.Text;

namespace Demo;

public class AuthService
{
    private readonly Dictionary<string, string> _users = new()
    {
        { "admin", Md5("admin123") }
    };

    public bool Login(string username, string password)
    {
        if (!_users.ContainsKey(username))
        {
            return false;
        }

        return _users[username] == Md5(password);
    }

    public string IssueToken(string username)
    {
        return $"token-{username}-12345";
    }

    private static string Md5(string input)
    {
        using var md5 = MD5.Create();
        var bytes = md5.ComputeHash(Encoding.UTF8.GetBytes(input));
        return Convert.ToHexString(bytes);
    }
}
